File: Algorithm/Leetcode/0000_0100/0021_0030/0021.py
# 1389. Create Target Array in the Given Order

"""https://leetcode.com/problems/create-target-array-in-the-given-order/discuss/560890/Python-3-in-one-line"""

class Solution:
    def createTargetArray(self, nums: List[int], index: List[int]) -> List[int]:    
        return functools.reduce(lambda r, p: r.insert(p[0], p[1]) or r, zip(index, nums), [])

File: Algorithm/Leetcode/0000_0100/0021_0030/0021.cpp
// 1389. Create Target Array in the Given Order

class Solution {
public:
    vector<int> createTargetArray(vector<int>& nums, vector<int>& index) {
		vector<int> output;
		output.reserve(nums.size());

		for (size_t i = 0; i < nums.size(); i++)
		{
			output.insert(output.begin() + index[i], nums[i]);
		}
		return output;
    }
};